pkg/azure/types/types.go

```diff
@@ -85,13 +92,64 @@ type AzureInterface struct {
 
 	// SecurityGroup is the security group associated with the interface
 	SecurityGroup string `json:"security-group,omitempty"`
+
+	// vmssName is the name of the virtual machine scale set. This field is
+	// set by extractIDs()
+	vmssName string
+
+	// vmName is the name of the virtual machine
+	vmName string
```

Comment:
mind renaming this to `vmID` ? I realize the setup is going to be a bit weird when we add support for plain VMs because in that case `vmssName` + `vmID` will be empty but we'll have a valid `vmName` and would like to deconflict now.

Reply:
But isn't this the name so we can reuse for VMs? I assume the vmID will be fully qualified.

```
subscriptions/xxx/resourceGroups/yyy/providers/Microsoft.Compute/virtualMachineScaleSets/ssss/virtualMachines/vvv/networkInterfaces/iii
```

`vmName` is currently et to `vvv`

Reply:
I think the nomenclature in azure is a bit different:
- for plain VMs the vmName is what you see in the console at the machine
- for VMSS the vmName will be `$vmssName-vvvvvv` (https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-instance-ids)

In this case for VMSS we treat the vmID to be vmName which does not align with how Azure wants to think about things.

Don't want to over index on this, we can always re-work this code a bit later.

Reply:
OK. That makes sense. I've renamed it.